python/hsfs/core/arrow_flight_client.py

```diff
@@ -0,0 +1,335 @@
+            feature_view, tds_version
+        )
+        path = (
+            f"{training_dataset_metadata['featurestore_name']}_Training_Datasets/"
+            f"{training_dataset_metadata['name']}_{training_dataset_metadata['version']}"
+            f"_{training_dataset_metadata['tds_version']}/"
+            f"{training_dataset_metadata['name']}_{training_dataset_metadata['version']}/"
+            f"{training_dataset_metadata['name']}_{training_dataset_metadata['version']}.parquet"
+        )
+        full_path = f"/Projects/{training_dataset_metadata['featurestore_name']}/{path}"
+        return full_path
+
+    def _training_dataset_metadata_from_feature_view(self, feature_view, tds_version):
+        training_dataset_metadata = {}
+        training_dataset_metadata["name"] = feature_view.name
+        training_dataset_metadata["version"] = f"{feature_view.version}"
+        training_dataset_metadata["tds_version"] = f"{tds_version}"
+        training_dataset_metadata["query"] = self._get_query_object(
+            feature_view.query, feature_view.query.to_string()
+        )
+        training_dataset_metadata[
+            "featurestore_name"
+        ] = feature_view.query._left_feature_group.feature_store_name.replace(
+            "_featurestore", ""
+        )
+
+        return training_dataset_metadata
+
+    def _get_query_object(self, query, query_str):
+        query_string = query_str.replace(
+            f"`{query._left_feature_group.feature_store_name}`.`",
+            f"`{query._left_feature_group.feature_store_name.replace('_featurestore','')}.",
+        ).replace("`", '"')
+
+        (
+            featuregroups,
+            features,
+            filters,
+        ) = self._collect_featuregroups_features_and_filters(query)
+
+        query = {
+            "query_string": query_string,
+            "featuregroups": featuregroups,
+            "features": features,
+            "filters": filters,
+        }
+        return query
+
+    def _update_features(self, features, fg_name, new_features):
+        updated_features = features.get(fg_name, set())
+        updated_features.update(new_features)
+        features[fg_name] = updated_features
+
+    def _collect_featuregroups_features_and_filters(self, query):
+        (
+            featuregroups,
+            features,
+            filters,
+        ) = self._collect_featuregroups_features_and_filters_rec(query)
+        filters = self._filter_to_expression(filters, featuregroups)
+        for feature in features:
+            features[feature] = list(features[feature])
+        return featuregroups, features, filters
+
+    def _collect_featuregroups_features_and_filters_rec(self, query):
+        featuregroups = {}
+        fg = query._left_feature_group
+        fg_name = f"{fg.feature_store_name.replace('_featurestore','')}.{fg.name}_{fg.version}"  # featurestore.name_version
```

Comment:
instead of always doing this `fg.feature_store_name.replace('_featurestore','')` in multiple places. i think it would be better to add a new function to FeatureGoupBase and instead call `fg.project_name` (it'll do this action)